processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -375,14 +375,16 @@ public Type erasure() {
      *
      * @return {@code true} if and only if this type is assignable to the given other type.
      */
-    // TODO This doesn't yet take wild card types into account; e.g. ? extends Integer wouldn't be assignable to Number
-    // atm.
     public boolean isAssignableTo(Type other) {
         if ( equals( other ) ) {
             return true;
         }
 
-        return typeUtils.isAssignable( typeMirror, other.typeMirror );
+        TypeMirror typeMirrotToMatch = isWildCardExtendsBound() ? getTypeBound().typeMirror : typeMirror;
+        TypeMirror otherTypeMirrotToMatch = other.isWildCardSuperBound() ?
```

Comment:
I think that there is no test that tests when `other` is `wildCardSuperBound`.  Can you add a case like that as well?